language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MappingContextProvider.java

```diff
@@ -187,12 +188,15 @@ protected boolean hasReadonlyKW(NonTerminalNode evalNodeAtCursor) {
         return this.expressionCompletions(context);
     }
 
-    protected NonTerminalNode getEvalNode(BallerinaCompletionContext context) {
+    protected Optional<Node> getEvalNode(BallerinaCompletionContext context) {
         NonTerminalNode nodeAtCursor = context.getNodeAtCursor();
```

Comment:
Can we remove this line and return `CommonUtil.getMatchingNode(context.getNodeAtCursor, predicate);`?